daemon/src/projection.rs

```diff
@@ -1,55 +1,82 @@
+use std::collections::HashMap;
+
 use crate::bitmex_price_feed::Quote;
 use crate::model::TakerId;
 use crate::{Cfd, Order, UpdateCfdProposals};
 use tokio::sync::watch;
 use xtra_productivity::xtra_productivity;
 
 pub struct Actor {
-    tx_cfds: watch::Sender<Vec<Cfd>>,
-    tx_order: watch::Sender<Option<Order>>,
-    tx_quote: watch::Sender<Quote>,
-    tx_settlements: watch::Sender<UpdateCfdProposals>,
-    // TODO: Use this channel to communicate maker status as well with generic
-    // ID of connected counterparties
-    tx_connected_takers: watch::Sender<Vec<TakerId>>,
+    tx: Tx,
+}
+
+pub struct Feeds {
+    pub cfds: watch::Receiver<Vec<Cfd>>,
+    pub order: watch::Receiver<Option<Order>>,
+    pub quote: watch::Receiver<Quote>,
+    pub settlements: watch::Receiver<UpdateCfdProposals>,
+    pub connected_takers: watch::Receiver<Vec<TakerId>>,
 }
 
 impl Actor {
-    pub fn new(
-        tx_cfds: watch::Sender<Vec<Cfd>>,
-        tx_order: watch::Sender<Option<Order>>,
-        tx_quote: watch::Sender<Quote>,
-        tx_settlements: watch::Sender<UpdateCfdProposals>,
-        tx_connected_takers: watch::Sender<Vec<TakerId>>,
-    ) -> Self {
-        Self {
-            tx_cfds,
-            tx_order,
-            tx_quote,
-            tx_settlements,
-            tx_connected_takers,
-        }
+    pub fn new(init_cfds: Vec<Cfd>, init_quote: Quote) -> (Self, Feeds) {
+        let (tx_cfds, rx_cfds) = watch::channel(init_cfds);
+        let (tx_order, rx_order) = watch::channel::<Option<Order>>(None);
+        let (tx_update_cfd_feed, rx_update_cfd_feed) =
+            watch::channel::<UpdateCfdProposals>(HashMap::new());
```

Comment:
Don't think you should need turbo fish here if you are directly assigning to fields.

Reply:
indeed! changed :)